### PR TITLE
Feat zone align

### PR DESCRIPTION
**Pull request by pauliuskalv**

### CODE REVIEW

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/Goal.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                {
+                    lostFrameCounter++;
+                    i++;
+                    continue;
+                }
+                else
+                {
+                    speeds[i] = Math.Sqrt( ( ( (point.X - lastPoint.X) * (point.X - lastPoint.X) ) * mulX + 
+                                            ( (point.Y - lastPoint.Y) * (point.Y - lastPoint.Y) ) * mulY ) );
+                    speeds[i] /= ((double)lostFrameCounter + 1.0f);
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/Goal.cs]: You cast `double` on `lostFrameCounter` and then add `float` value, should't it be a double one aswell?

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/PositionChecker.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        /// <summary>
+        /// Defines the goal zones, which hold the point of no return for the ball
+        /// </summary>
+        public RectF zoneOne;
+        public RectF zoneTwo;
+
+        /// <summary>
+        /// Holds the goals, which occured during the session
+        /// </summary>
+        private Queue<Goal> goals;
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/PositionChecker.cs]: Same naming convention with `_`.

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/PositionChecker.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                                CurrentEvent currentEvent, Action<int,int,CurrentEvent> setter,
+                                EventHandler<EventArgs> GoalEvent, Queue<PointF> ballCoordinates)
+        {
+            // Check if this particular point signals that the ball is lost
+            if (lastBallCoordinates == null)
+            {
+                if (framesLost == GoalFramesToCountGoal)
+                {
+                    // It is, so check if a goal is about to occur
+                    if (ballInFirstGoalZone)
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/PositionChecker.cs]: if `ballInFirstGoalZone == true`, then `ballInSecondGoalZone == false`, right? Perhaps you could make them attributes and create custom setter when one of them becomes `true`, the other one becomes `false` automatically?

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/PositionChecker.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                    ballInSecondGoalZone = false;
+                }
+            }
+
+            return;
+        }
+
+        public double calculateSpeed(PointF one, PointF two, EventHandler<EventArgs> PositionEvent)
+        {
+            if (one != null && two != null)
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/PositionChecker.cs]: If am not mistaken, you could do something like this instead: 
`return Math.Sqrt( (one?.X - two?.X) * (one?.X - two?.X)  + (one?.Y - two?.Y) * (one?.Y - two?.Y) ) ?? 0;`

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        /// Defines the rows of foosmen
+        /// </summary>
+        public RectF[] rows;
+
+        /// <summary>
+        /// Defines the row the ball is currently in
+        /// </summary>
+        public Row currentRow;
+
+        private int RedGoalieZone;
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs]: Same naming conventions with `_`.

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        private int RedDefenceZone;
+        private int BlueAttackZone;
+        private int RedMidfieldZone;
+        private int BlueMidfieldZone;
+        private int RedAttackZone;
+        private int BlueDefenceZone;
+        private int BlueGoalieZone;
+
+        public void checkRow(PointF lastBallCoordinates, ref CurrentEvent currentEvent)
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs]: Methods start from **Capital** letter.

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                    break;
+                }
+            }
+        }
+
+        public void CalculateRows(System.Drawing.Rectangle tableZone)
+        {
+            // Declare constants
+            float[] multipliers = new float[8];
+            multipliers[0] = 0.1158f;
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs]: Magic numbers. Should probably go to config file.

File: FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+            multipliers[5] = 0.1348f;
+            multipliers[6] = 0.1832f;
+            multipliers[7] = 0.2124f;
+
+            rows = new RectF[8];
+
+            rows[0] = new RectF(tableZone.Left, tableZone.Top,
+                                tableZone.Right, tableZone.Top + (tableZone.Height * multipliers[0]));
+
+            for (int i = 1; i < 8; i++)
```
Comment (ramumnesys) [inline FoosLiveAndroid/FoosLiveAndroid/Util/GameControl/RowChecker.cs]: Comments would be handy here.

File: FoosLiveAndroid/FoosLiveAndroid/Util/Detection/ColorDetector.cs [hunk 9]
```diff
@@ -188,6 +125,8 @@ public bool DetectBall(Hsv ballHsv, out Rectangle rect, out Rectangle blobBox)
             Hsv upperLimit = new Hsv(ballHsv.Hue + Threshold, ballHsv.Satuation + Threshold, ballHsv.Value + Threshold);
 
             Image<Gray, byte> imgFiltered = image.InRange(lowerLimit, upperLimit);
+            imgFiltered.Erode(1);
```
[the remark this answers is not in the document]
Comment (pauliuskalv) [reply FoosLiveAndroid/FoosLiveAndroid/Util/Detection/ColorDetector.cs]: :(

**Pull request merged**